tests/phpunit/tests/test-flags.php

```diff
@@ -23,6 +23,12 @@ function setUp() {
 		self::$polylang = new PLL_Frontend( self::$polylang->links_model );
 	}
 
+	function test_default_flag() {
+		$lang = self::$polylang->model->get_language( 'en' );
+		$this->assertEquals( plugins_url( '/flags/us.png', POLYLANG_FILE ), $lang->get_display_flag_url() ); // Bug fixed in 2.8.1.
+		$this->assertEquals( 1, preg_match( '#<img src="data:image\/png;base64,(.+)" title="English" alt="English" width="16" height="11" style="(.+)" \/>#', $lang->get_display_flag() ) );
```

Comment:
You have the [assertStringMatchesFormat()](https://phpunit.de/manual/5.7/en/appendixes.assertions.html#appendixes.assertions.assertStringMatchesFormat) assertions if you want to test with regexps.

Reply:
If you are thinking to https://github.com/polylang/polylang/pull/559/files#diff-34d98b479a021607a28a4070fb0bea86R241-R252, my goal was not to avoid duplicating `sprintf()` calls but to find a way to display the inline style when we have a height and/or width and noting wen we have nothing. I chose this way rather than making several nested tests.

I didn't know  `assertStringMatchesFormat`. Useful.